tests/net/route_mcast/src/main.c

```diff
@@ -0,0 +1,665 @@
+					0, 0, 0, 0,
+					0, 0, 0, 0,
+					0, 0, 0, 0 } } };
+static struct in6_addr mcast_prefix_global = { { {
+					0xFF, 0x0E, 0, 0,
+					0, 0, 0, 0,
+					0, 0, 0, 0,
+					0, 0, 0, 0 } } };
+/*
+ * full network prefix based address,
+ * see RFC-3306 for details
+ * FF3F:40:FD01:101:: \128
+ * network prefix FD01:101::\64
+ */
+static struct in6_addr mcast_prefix_nw_based = { { {
+		0xFF, 0x3F, 0, 0x40,
+		0xFD, 0x01, 0x01, 0x01,
+		0, 0, 0, 0,
+		0, 0, 0, 0 } } };
+
+static uint8_t forwarding_counter;
+static bool iface_1_forwarded;
+static bool iface_2_forwarded;
+static bool iface_3_forwarded;
+
+struct net_route_mcast_scenario_cfg {
+	struct in6_addr src;
+	struct in6_addr mcast;
+	bool is_active;
+};
+
+static struct net_route_mcast_scenario_cfg active_scenario;
+
+int net_route_mcast_dev_init(struct device *dev)
+{
+	return 0;
+}
+
+static uint8_t *net_route_mcast_get_mac(struct device *dev)
+{
+	struct net_route_mcast_iface_cfg *cfg = dev->data;
+
+	if (cfg->mac_addr[2] == 0x00) {
+		/* 00-00-5E-00-53-xx Documentation RFC 7042 */
+		cfg->mac_addr[0] = 0x00;
+		cfg->mac_addr[1] = 0x00;
+		cfg->mac_addr[2] = 0x5E;
+		cfg->mac_addr[3] = 0x00;
+		cfg->mac_addr[4] = 0x53;
+		cfg->mac_addr[5] = sys_rand32_get();
+	}
+
+	cfg->ll_addr.addr = cfg->mac_addr;
+	cfg->ll_addr.len = 6U;
+
+	return cfg->mac_addr;
+}
+
+static void net_route_mcast_add_addresses(struct net_if *iface,
+		struct in6_addr *ipv6, struct in6_addr *ll_addr)
+{
+	struct net_if_mcast_addr *maddr;
+	struct net_if_addr *ifaddr;
+
+	uint8_t *mac = net_route_mcast_get_mac(net_if_get_device(iface));
+
+	net_if_set_link_addr(iface, mac, sizeof(struct net_eth_addr),
+				     NET_LINK_ETHERNET);
+
+	ifaddr = net_if_ipv6_addr_add(iface, ipv6, NET_ADDR_MANUAL, 0);
+	zassert_not_null(ifaddr, "Cannot add global IPv6 address");
+
+	ifaddr->addr_state = NET_ADDR_PREFERRED;
+
+	ifaddr = net_if_ipv6_addr_add(iface, ll_addr, NET_ADDR_MANUAL, 0);
+	zassert_not_null(ifaddr, "Cannot add ll IPv6 address");
+
+	ifaddr->addr_state = NET_ADDR_PREFERRED;
+
+	maddr = net_if_ipv6_maddr_add(iface, &in6addr_mcast);
+	zassert_not_null(maddr, "Cannot add multicast IPv6 address");
+}
+
+static void net_route_mcast_iface_init1(struct net_if *iface)
+{
+	iface_1 = iface;
+	net_route_mcast_add_addresses(iface, &iface_1_addr, &ll_addr_1);
+}
+
+static void net_route_mcast_iface_init2(struct net_if *iface)
+{
+	iface_2 = iface;
+	net_route_mcast_add_addresses(iface, &iface_2_addr, &ll_addr_2);
+}
+
+static void net_route_mcast_iface_init3(struct net_if *iface)
+{
+	iface_3 = iface;
+	net_route_mcast_add_addresses(iface, &iface_3_addr, &ll_addr_3);
+}
+
+static bool check_packet_addresses(struct net_pkt *pkt)
+{
+	struct net_ipv6_hdr *ipv6_hdr = NET_IPV6_HDR(pkt);
+
+	if ((memcmp(&active_scenario.src,
+			&ipv6_hdr->src,
+			sizeof(struct in6_addr)) != 0) ||
+			(memcmp(&active_scenario.mcast,
+					&ipv6_hdr->dst,
+					sizeof(struct in6_addr)) != 0)) {
+		return false;
+	}
+
+	return true;
+}
```

Comment:
empty line after }

Reply:
Done. Not hit by automated check cause after the marked line.